ansible/roles/slurm_headnode/templates/add_user.py

```diff
@@ -44,16 +44,27 @@ def get_ldap_connection(
     return conn
 
 
-def get_next_uid(min_allowed: int = 10000, max_allowed: int = 20000) -> int:
-    # Get user-id to use
-    users = getpwall()
-    users = list(map(lambda x: x.pw_uid, users))
-    users = list(filter(lambda x: min_allowed <= x <= max_allowed, users))
-    if not users:
-        return min_allowed
-    proposed_uid = max([max(users) + 1, min_allowed])
-    assert min_allowed <= proposed_uid <= max_allowed
-    return proposed_uid
+def get_users_from_ldap(ldap_connection: Connection, organizational_unit: str | None = None) -> list[Entry]:
+    if not organizational_unit:
+        ldap_connection.search(
+            search_base=LDAP_SEARCH_BASE,
+            search_filter=f'(objectclass=posixAccount)',
+            attributes=ALL_ATTRIBUTES,
+        )
+        return ldap_connection.entries[:]
+    ldap_connection.search(
```

Comment:
Only suggestion would be to add a blank line here maybe after the if not. Since it's easy to miss that the indents changed. Being super pedantic though